packages/vite/src/node/utils.ts

```diff
@@ -204,6 +204,9 @@ export function prettifyUrl(url: string, root: string): string {
 export function isObject(value: unknown): value is Record<string, any> {
   return Object.prototype.toString.call(value) === '[object Object]'
 }
+export function isDefined<T>(value: T | undefined | null): value is T {
```

Comment:
**suggestion:** I think if the input would be e.g. `number | null` the result could be also `number | null`  
Maybe we need to use something like `Omit` here 🤔 


---

Please also add a blank line above this function 🙂

Reply:
>  I think if the input would be e.g. number | null the result could be also number | null. Maybe we need to use something like Omit here 🤔

TypeScript already does it, see https://www.typescriptlang.org/play?ssl=9&ssc=1&pln=9&pc=2#code/MYewdgzgLgBGCMAuOBXAtgIwKYCcYF4Z4BmYgdgChRJYwAmZMFAG2YNVavGjmMfWw4APk1bsAsgEMoACwB0OSWAAmINAAoAlAB4ADHICsMAPwwALHRj9O1HswCW0ZOqaZcIls00BtALrtvBAAaODoQsGJfLhoYB2gAMXtmKFwsZXY4qDkAMySUnHVHABEsXLA0zQoKbJQwYCh7cBhi0vty5W0AFQA+dQA3SWYULGROmCEYWuVW9vGOL2QBoaxmiBgxgG8KGBgcLCgUHDAYJeGYAEJ8QimZtJgAMnuTwbPLwlFmCgBfIA and look how `listFiltered` is correctly typed `number[]` and not `(number|null)[]`.

> Please also add a blank line above this function 🙂

Done :-).

Reply:
> Done :-).

I can't update this PR since it's already merged; created a new one over there https://github.com/vitejs/vite/pull/3257